Utils/utils.py

```
@@ -482,6 +482,25 @@ def proxy_request(method: str, url: str, headers=None, timeout=TIMEOUT, bypass_p
     return res
 
 
+def resolve_redirect_url(url: str, timeout: int = 5) -> str | None:
+    """
+    Resolves a redirect URL to its final destination URL.
+    """
+    try:
+        res = proxy_request("HEAD", url, timeout=timeout, allow_redirects=True, bypass_proxy=True)
+        return res.url
+
+    except Exception:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)